packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -3791,11 +3792,26 @@ export class ProjectService extends BaseService {
 
         if (groupByColumns && groupByColumns.length > 0) {
             // Wrap the original query in a CTE
-            let pivotedSql = `WITH original_query AS (${userSql}), group_by_query AS (${groupByQuery}), pivot_query AS (${pivotQuery})`;
-
-            pivotedSql += `\nSELECT * FROM pivot_query WHERE ${q}row_index${q} <= ${
+            const totalColumnsQuery = `
+                SELECT COUNT(DISTINCT ${groupByColumns
+                    .map((col) => `${q}${col.reference}${q}`)
+                    .join(', ')}
+            `;
+            let pivotedSql = `
+            WITH original_query AS (${userSql}), 
+                 group_by_query AS (${groupByQuery}), 
+                 pivot_query AS (${pivotQuery}),
+                 total_columns AS (${totalColumnsQuery}) as total_columns
+            FROM group_by_query
+            )`;
+
+            // Add a max column limit to avoid too many columns and performance issues.
+            // Warn the user if we exceed it.
+
+            pivotedSql += `\nSELECT p.*, t.total_columns FROM pivot_query p CROSS JOIN total_columns t WHERE p.${q}row_index${q} <= ${
```

Comment:
adds total_columns per row
I didn't want to make this a separate query so this cross join helped. lmk if you think otherwise